Clean/Circomlib/Sign.lean

```diff
@@ -0,0 +1,61 @@
+import Clean.Circuit
+import Clean.Utils.Field
+import Clean.Utils.Bits
+import Clean.Circomlib.CompConstant
+
+/-
+Original source code:
+https://github.com/iden3/circomlib/blob/master/circuits/sign.circom
+
+The original Sign circuit uses a specific constant for the BN128 curve.
+We generalize this to work with any prime field by using (p-1)/2.
+-/
+
+namespace Circomlib
+variable {p : ℕ} [Fact p.Prime] [Fact (p < 2^254)] [Fact (p > 2^253)]
+
+namespace Sign
+/-
+template Sign() {
+    signal input in[254];
+    signal output sign;
+
+    component comp = CompConstant(10944121435919637611123202872628637544274182200208017171849102093287904247808);
+
+    var i;
+
+    for (i=0; i<254; i++) {
+        comp.in[i] <== in[i];
+    }
+
+    sign <== comp.out;
+}
+-/
+
+def main (input : Vector (Expression (F p)) 254) :=
+  -- Use (p-1)/2 as the constant for comparison
+  CompConstant.circuit ((p - 1) / 2) input
+
+def circuit : FormalCircuit (F p) (fields 254) field where
+  main
+  localLength input := (CompConstant.circuit ((p - 1) / 2)).localLength input
+  localLength_eq := by sorry
+  subcircuitsConsistent := by simp +arith [circuit_norm, main]
```

Comment:
for simple circuits like these, you can rely on the default tactics to prove these theorems
(the default tactics should work for pretty much any circuit that doesn't use loops)
```suggestion
  localLength input := (CompConstant.circuit ((p - 1) / 2)).localLength input
```